staging/src/k8s.io/apiserver/pkg/util/dryrun/dryrun.go

```diff
@@ -16,7 +16,32 @@ limitations under the License.
 
 package dryrun
 
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
 // IsDryRun returns true if the DryRun flag is an actual dry-run.
 func IsDryRun(flag []string) bool {
 	return len(flag) > 0
 }
+
+func RevertDryRunMetadata(originalObj, newObj runtime.Object) error {
+	originalObjMeta, err := meta.Accessor(originalObj)
+	if err != nil {
+		return errors.NewInternalError(err)
+	}
+	newObjMeta, err := meta.Accessor(newObj)
+	if err != nil {
+		return errors.NewInternalError(err)
+	}
+	if originalObjMeta.GetGenerateName() != "" {
+		newObjMeta.SetName("")
```

Comment:
```suggestion
		// If both the generateName and name fields are non-empty strings, then non-dry-run creation will fail.
		newObjMeta.SetName("")
```

Reply:
I think this text better explains. WDYT?